src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,559 @@
+      revert PPT_TotalCostOverTolerance();
+    }
+    (, uint strike1,) = OptionEncoding.fromSubId(tradeData.trades[0].subId.toUint96());
+    (, uint strike2,) = OptionEncoding.fromSubId(tradeData.trades[1].subId.toUint96());
+    uint markValueToStrikeDiffRatio = ((strike2 - strike1).divideDecimal((tradePrices[1] - tradePrices[0])));
+    if (
+      markValueToStrikeDiffRatio < $.ppParams.minMarkValueToStrikeDiffRatio
+        || markValueToStrikeDiffRatio > $.ppParams.maxMarkValueToStrikeDiffRatio
+    ) {
+      revert PPT_MarkValueNotWithinBounds();
+    }
+    return totalCost;
+  }
+
+  function _getCallPrice(IRfqModule.TradeData memory trade) internal view returns (uint callPrice) {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    (uint expiry, uint strike, bool isCall) = OptionEncoding.fromSubId(trade.subId.toUint96());
+    if (!isCall) {
+      revert PPT_OnlyShortCallsAllowed();
+    }
+    if (block.timestamp >= expiry) {
+      revert PPT_OptionExpired();
+    }
+    uint timeToExpiry = expiry - block.timestamp;
+    if (timeToExpiry < $.ppParams.optionMinTimeToExpiry || timeToExpiry > $.ppParams.optionMaxTimeToExpiry) {
+      revert PPT_OptionExpiryOutOfBounds();
+    }
+
+    (uint vol, uint forwardPrice) = _getFeedValues(strike.toUint128(), expiry.toUint64());
+
+    (callPrice,,) = Black76.pricesAndDelta(
+      Black76.Black76Inputs({
+        timeToExpirySec: timeToExpiry.toUint64(),
+        volatility: (vol.multiplyDecimal($.ppParams.optionVolSlippageFactor)).toUint128(),
+        fwdPrice: forwardPrice.toUint128(),
+        strikePrice: strike.toUint128(),
+        discount: 1e18
+      })
+    );
+    return callPrice;
+  }
+
+  function _getFeedValues(uint128 strike, uint64 expiry) internal view returns (uint vol, uint forwardPrice) {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    StandardManager srm = StandardManager(address(getBaseTSAAddresses().manager));
+    IStandardManager.AssetDetail memory assetDetails = srm.assetDetails($.optionAsset);
+    (, IForwardFeed fwdFeed, IVolFeed volFeed) = srm.getMarketFeeds(assetDetails.marketId);
+    (vol,) = volFeed.getVol(strike, expiry);
+    (forwardPrice,) = fwdFeed.getForwardPrice(expiry);
+  }
+
+  ///////////////////
+  // Account Value //
+  ///////////////////
+
+  function _getSubAccountStats() internal view returns (int tradeBalance, uint baseBalance, int cashBalance) {
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+    ISubAccounts.AssetBalance[] memory balances = tsaAddresses.subAccounts.getAccountBalances(subAccount());
+    for (uint i = 0; i < balances.length; i++) {
+      if (balances[i].asset == _getPPTSAStorage().optionAsset) {
+        // Note: switching this to int since we will have a negative balance from buying ITM options.
+        // possibly offsetting slightly by selling slightly higher strike price options.
+        // Is this right?
+        tradeBalance += balances[i].balance;
```

Comment:
balance is a pure number of calls. So this will always equal 0 (negatives will cancel out against positives)

Reply:
Oh, really? Nice. I wasn't sure exactly what `optionAsset was and if that was just some wrapper over LRTs used or something. I can clean up a lot of stuff here then.